### DISCUSSION TITLE
Literatürde Göze Çarpanı Buraya Atalım!

### DISCUSSION THREAD

**Discussion by sertacates** [iflGARAJI10100/taMAM#20] [Ideas]
Gençler, çalışmalarımıza kaynaklık eden literatür bilgisini ISSUES altında Kaynakça olarak zaten yapacağız ama buraya da gözden geçirdiğimiz her şeyi atalım. İlla kaynakçaya ekleyeceklerimiz değil tüm taradıklarımız bu kısımda gözüksün.

**Comment by S-Toprak** [comment]
Hocam şu kaynakları buldum: 
https://github.com/nilansaha/NLP-Search-Engine
https://github.com/Hermann-web/Search-engine-with-python-nlp
bu kaynaklar projemize yardımcı olur diye düşünüyorum

**Comment by ErAk042** [comment]
https://github.com/gautamdasika/Document-Search-Engine
Bende bunu buldum incelemekte fayda var gibi.

**Comment by ErAk042** [comment]
https://link.springer.com/referenceworkentry/10.1007/978-3-319-32001-4_182-1
https://www.lexalytics.com/blog/context-analysis-nlp/
bulduğum birkaç kaynak daha

**Comment by ErAk042** [comment]
https://monkeylearn.com/blog/text-cleaning/
https://towardsdatascience.com/nlp-building-text-cleanup-and-preprocessing-pipeline-eba4095245a0
https://monkeylearn.com/text-analysis/

**Comment by ErAk042** [comment]
https://stackoverflow.com/questions/12118720/python-tf-idf-cosine-to-find-document-similarity

sklearn ve nltk için güzel kodlar var. Hem benzerliği hem metin çıkarımını çok güzel izah ediyor

**Comment by ErAk042** [comment]
https://www.machinelearningplus.com/nlp/gensim-tutorial/

Bu kaynak tf-idfi, lda, lsi, word2vec do2vec için güzel ayrıca benzerlik metriklerine de deyinmiş

**Comment by ErAk042** [comment]
Konu ile alakalı değil ama biraz makine öğrenmesi 
https://github.com/keighrim/python-nlp-ml-tutorial/blob/master/NLTK.ipynb
https://www.lexalytics.com/blog/machine-learning-natural-language-processing/
https://www.bogotobogo.com/python/scikit-learn/scikit_machine_learning_Support_Vector_Machines_SVM_2.php

**Comment by ErAk042** [comment]
Biraz da doğal dil işleme temellerinden kaynak paylaşmak istiyorum
https://ybsansiklopedi.com/wp-content/uploads/2015/09/ybs_ddi_nlp.pdf
https://dergipark.org.tr/tr/download/article-file/1573501
Burada çok güzel bir çalışma var. Açık kaynak kodlu kütüphaneleri özellikle doğal dil işlemede kullanılanları tanıtıyor

**Comment by ErAk042** [comment]
Videolarda vardı, bizim alanla çok ilgili değil ama Kemal hocanın güzel bir kaynağı
https://dergipark.org.tr/en/download/article-file/207207
Kelime torbası yöntemi için Türkçe ve güzel bir makale
https://dergipark.org.tr/en/download/article-file/552012

Arama kısmında bunu kullanabiliriz belki yani arama motoru kısmında. Kavramlar Arası WordNet Tabanlı Anlamsal Benzerlik Değerlerinin
Farklı Metriklerle Değerlendirilmesi https://dergipark.org.tr/en/download/article-file/1374800
https://ieeexplore.ieee.org/abstract/document/6111017 bunun tam halini bulsak yararlı olur diye düşünüyorum

**Comment by ErAk042** [comment]
Veri madenciliğinde kullanabileceğimiz çeşitli araçlar https://www.researchgate.net/profile/Vijayarani-Mohan/publication/329800669_Text_Mining_Open_Source_Tokenization_Tools_An_Analysis/links/5e4d03d4299bf1cdb935885a/Text-Mining-Open-Source-Tokenization-Tools-An-Analysis.pdf

**Comment by sertacates** [comment]
https://github.com/iflGARAJI10100/taMAM/blob/main/aramaProjesiKategoriBulmaYonetmelik.ipynb bu dosyadaki kaynakları bu kısımdan da paylaşsaydık iyi olurdu. Karmaşıklık Matrisini ve scikit-learn kütüphanesini bildiğinizi ve kullandığınızı bu kısımdan önce belirtsek sonra kullansanız daha uygun olurdu. Neyse sıkıntı yok, ben sizin için paylaşıyorum. Bu arada ipynb güzel olmuş, elinize sağlık.

Kaynak 1: https://www.sciencedirect.com/topics/engineering/confusion-matrix#:~:text=A%20confusion%20matrix%20is%20a,malignant%20tissue%20is%20considered%20cancerous.

Kaynak 2: https://en.wikipedia.org/wiki/Confusion_matrix

Kaynak 1: https://www.w3schools.com/python/python_ml_confusion_matrix.asp#:~:text=What%20is%20a%20confusion%20matrix,the%20predictions%20we%20have%20made.

Kaynak 2: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.ConfusionMatrixDisplay.html

Kaynak 3: https://scikit-learn.org/stable/modules/model_evaluation.html

**Comment by sertacates** [comment]
https://scikit-learn.org/stable/user_guide.html#user-guide

**Comment by sertacates** [comment]
https://ashutoshtripathi.com/2020/09/04/word2vec-and-semantic-similarity-using-spacy-nlp-spacy-series-part-7/
